dapr_agents/agents/durableagent/agent.py

```diff
@@ -92,24 +107,30 @@ def model_post_init(self, __context: Any) -> None:
         }
         self.register_agentic_system()
 
-    async def run(self, input_data: Optional[Union[str, Dict[str, Any]]] = None) -> Any:
+    async def run(self, input_data: Union[str, Dict[str, Any]]) -> Any:
         """
-        Run the durable agent with the given input.
-        TODO: For DurableAgent, this method should trigger the workflow execution maybe..?
+        Fire up the workflow, wait for it to complete, then return the final serialized_output.
 
         Args:
-            input_data: The input data for the agent to process.
-
+            input_data (Union[str, Dict[str, Any]]): The input for the workflow. Can be a string (task) or a dict.
         Returns:
-            The result of the workflow execution.
+            Any: The final output from the workflow execution.
         """
-        # TODO: For DurableAgent, the run method should trigger the workflow
-        logger.info(
-            f"DurableAgent {self.name} run method called with input: {input_data}"
-        )
+        # Make sure the Dapr runtime is running
+        if not self.wf_runtime_is_running:
+            self.start_runtime()
 
-        # Return a message indicating this is a durable agent and agent start via run for durable agent is yet to be determined.
-        return f"DurableAgent {self.name} is designed to run as a workflow service asynchronously. Use .as_service() and/or .start() instead for now. The workflow endpoints can also be usedto interact with this agent."
+        # Prepare input payload for workflow
+        if isinstance(input_data, dict):
+            input_payload = input_data
+        else:
+            input_payload = {"task": input_data}
+
+        # Kick off the workflow and block until it finishes:
+        return await self.run_and_monitor_workflow_async(
+            workflow=self._workflow_name,
+            input=input_payload,
```

Comment:
i see now, thank you for implementing this! So now both agent types can be started with .as_service() and run()?

Reply:
`DurableAgent` can be `as_service()` and `run()`. However, `Agent` is only `run()`.